src/chainstate/stacks/db/blocks.rs

```diff
@@ -4888,9 +4935,9 @@ impl StacksChainState {
     /// block's transactions.  Finally, it returns the execution costs for the microblock stream
     /// and for the anchored block (separately).
     /// Returns None if we're out of blocks to process.
-    fn append_block(
+    fn append_block<'a>(
```

Comment:
Can you update the docstring to describe why returning a `PreCommitClairtyBlock<'a>` is necessary here?  Also, I think the docstring is obsolete in that it describes all the things that are now currently encapsulated in the `StacksEpochReceipt`.